controllers/managedcluster.go

```diff
@@ -75,15 +87,16 @@ func (r *MultiClusterHubReconciler) ensureKlusterletAddonConfig(m *operatorsv1.M
 		return ctrl.Result{}, err
 	}
 
-	klusterletaddonconfig := getKlusterletAddonConfig()
+	newKlusterletaddonconfig := getKlusterletAddonConfig(m)
+	klusterletaddonconfig := newKlusterletaddonconfig.DeepCopy()
 	nsn := types.NamespacedName{
 		Name:      KlusterletAddonConfigName,
 		Namespace: ManagedClusterName,
 	}
+
 	err = r.Client.Get(ctx, nsn, klusterletaddonconfig)
 	if err != nil && errors.IsNotFound(err) {
```

Comment:
You'll need to return the error if `err != nil` and it's not because it wasn't found.

```suggestion
	if errors.IsNotFound(err) {
```